robot_controllers_interface/src/controller_manager.cpp

```diff
@@ -184,65 +184,118 @@ int ControllerManager::requestStop(const std::string& name)
 void ControllerManager::update(const ros::Time& time, const ros::Duration& dt)
 {
   // Reset handles
-  for (JointHandleList::iterator j = joints_.begin(); j != joints_.end(); j++)
-    (*j)->reset();
+  for (const auto& joint_handle: joints_)
+    joint_handle->reset();
 
   // Update controllers
-  for (ControllerList::iterator c = controllers_.begin(); c != controllers_.end(); c++)
+  for (const auto& c: controllers_)
   {
-    (*c)->update(time, dt);
+    c->update(time, dt);
   }
 }
 
 void ControllerManager::reset()
 {
   // Update controllers
-  for (ControllerList::iterator c = controllers_.begin(); c != controllers_.end(); c++)
+  for (const auto& c: controllers_)
   {
-    (*c)->reset();
+    c->reset();
   }
 }
 
-bool ControllerManager::addJointHandle(JointHandlePtr& j)
+bool ControllerManager::addJointHandle(JointHandlePtr& joint_handle_ptr)
 {
-  // TODO: check for duplicate names?
-  joints_.push_back(j);
+  // If we don't already have a handle with
+  // a duplicate name, we add it
+
+  for (const auto& joint_handle: joints_)
+  {
+    if (joint_handle->getName() == joint_handle_ptr->getName())
+    return false;
+  }
+  
+  joints_.push_back(joint_handle_ptr);
+  return true;
+}
+
+bool ControllerManager::addGyroHandle(GyroHandlePtr gyro_handle_ptr)
+{
+  // If we don't already have a handle with
+  // a duplicate name, we add it
+
+  for (const auto& gyro_handle: gyros_)
+  {
+    if (gyro_handle->getName() == gyro_handle_ptr->getName())
+    return false;
+  }
+
+  gyros_.push_back(gyro_handle_ptr);
   return true;
 }
 
 HandlePtr ControllerManager::getHandle(const std::string& name)
 {
   // Try joints first
-  for (JointHandleList::iterator j = joints_.begin(); j != joints_.end(); j++)
+  for (auto& j: joints_)
   {
-    if ((*j)->getName() == name)
-      return *j;
+    if (j and j->getName() == name)
+    {
+      return j;
+    }
+  }
+
+  // Then gyros
+  for (auto& gyro_handle_pointer: gyros_)
+  {
+    if (gyro_handle_pointer and gyro_handle_pointer->getName() == name)
+    {
+      return gyro_handle_pointer;
+    }
   }
 
   // Then controllers
-  for (ControllerList::iterator c = controllers_.begin(); c != controllers_.end(); c++)
+  for (auto& controller: controllers_)
   {
-    if ((*c)->getController()->getName() == name)
-      return (*c)->getController();
+    if (controller and controller->getController()->getName() == name)
```

Comment:
I does seem weird that we would allow / want null pointers to joint/gyros to be added.   Maybe just throw an exception in addJoint and addGryo if pointer is null.
@mikeferguson  is there a reason class loader would need/want to add null pointers?

Reply:
no reason to support null points - hence why I never checked them (although, yeah, we should probably check when adding them...)